zerver/models.py

```diff
@@ -653,32 +654,40 @@ class Realm(models.Model):  # type: ignore[django-manager-missing] # django-stub
 
     GIPHY_RATING_OPTIONS = {
         "disabled": {
-            "name": "GIPHY integration disabled",
+            "name": gettext_lazy("GIPHY integration disabled"),
             "id": 0,
         },
         # Source: https://github.com/Giphy/giphy-js/blob/master/packages/fetch-api/README.md#shared-options
         "y": {
-            "name": "Allow GIFs rated Y (Very young audience)",
+            "name": gettext_lazy("Allow GIFs rated Y (Very young audience)"),
             "id": 1,
         },
         "g": {
-            "name": "Allow GIFs rated G (General audience)",
+            "name": gettext_lazy("Allow GIFs rated G (General audience)"),
             "id": 2,
         },
         "pg": {
-            "name": "Allow GIFs rated PG (Parental guidance)",
+            "name": gettext_lazy("Allow GIFs rated PG (Parental guidance)"),
             "id": 3,
         },
         "pg-13": {
-            "name": "Allow GIFs rated PG13 (Parental guidance - under 13)",
+            "name": gettext_lazy("Allow GIFs rated PG13 (Parental guidance - under 13)"),
             "id": 4,
         },
         "r": {
-            "name": "Allow GIFs rated R (Restricted)",
+            "name": gettext_lazy("Allow GIFs rated R (Restricted)"),
             "id": 5,
         },
     }
 
+    def get_giphy_rating_options(self) -> Dict[str, Dict[str, object]]:
+        giphy_rating_options = deepcopy(self.GIPHY_RATING_OPTIONS)
+
+        for rating_type in giphy_rating_options.values():
+            rating_type["name"] = str(rating_type["name"])
+
+        return giphy_rating_options
```

Comment:
Hmm, is the deepcopy needed because this is a singleton, and the first time a given Python process renders the string, it resolves the lazy object into an actual string? Probably deserves a comment.

(Also, if so, does that issue apply to other translateable strings we define in `models.py` via `gettext_lazy`? I'm wondering if we're doing it wrong...).

Reply:
> Hmm, is the deepcopy needed because this is a singleton, and the first time a given Python process renders the string, it resolves the lazy object into an actual string? Probably deserves a comment.

I am not sure I understand what you mean. I used deepcopy to avoid modifying the original GIPHY_RATING_OPTIONS variable. However, it's possible that we could modify the original variable to make deepcopy unnecessary. Either way, my code works with or without deepcopy.

If we talk about this approach, I took `get_default_external_accounts()` as an example in `events.py` file

Reply:
Could this function just be this?
```
return [dict(name=str(option["name"]), id=option["id"]) for option in giphy_rating_options]
```